### PR TITLE
Freeze graphs using a loop over mutator methods.

### PR DESCRIPTION
**Pull request by jfinkels**

### CODE REVIEW

File: networkx/classes/function.py [hunk 1]
```diff
@@ -23,6 +23,11 @@
            'common_neighbors', 'is_weighted','is_negatively_weighted',
            'is_empty']
 
+#: This list includes the names of all methods that mutate a graph.
+MUTATORS = ['add_edge', 'add_edges_from', 'add_node', 'add_nodes_from',
```
Comment (harlowja) [inline networkx/classes/function.py]: Make this a tuple/frozenset to denote that its supposed to be immuatble?

File: networkx/classes/function.py [hunk 2]
```diff
@@ -186,16 +191,9 @@ def freeze(G):
...
-    G.remove_node=frozen
-    G.remove_nodes_from=frozen
-    G.add_edge=frozen
-    G.add_edges_from=frozen
-    G.remove_edge=frozen
-    G.remove_edges_from=frozen
-    G.clear=frozen
-    G.frozen=True
+    for method in MUTATORS:
+        setattr(G, method, frozen)
```
Comment (harlowja) [inline networkx/classes/function.py]: Should we also do ``hasattr`` and blowup (raise an attribute error), before setting random attributes, just a safety thing, but might be useful?

Comment (jfinkels) [inline networkx/classes/function.py]: Seems unnecessary to me. The attributes are not "random", nor are they arbitrary. They are fixed, and if any of the method names change, unit tests will catch such an error.

File: networkx/classes/function.py (hunk 1 above)
Comment (jfinkels) [inline networkx/classes/function.py]: Yes.

File: networkx/classes/function.py (hunk 2 above)
Comment (harlowja) [inline networkx/classes/function.py]: Ok dokie.

**Comment by harlowja** [comment]
:+1:  from me then.

**Comment by hagberg** [comment]
We're doing to drop this feature, #1698.

**Pull request closed without merging**